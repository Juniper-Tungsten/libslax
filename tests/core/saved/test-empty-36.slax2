version 1.2;

ns jcs extension = "http://xml.juniper.net/junos/commit-scripts/1.0";

param $how = "and";

main <op-script-results> {
    expr jcs:output("This " _ $how _ " that");
    
    call test($one = <one>);
    call test($one = <one> "two");
    call test($one = <one> {
        <two> "two";
    });
}

template test ($one = "null") {
    <output> {
        copy-of $one;
    }
}
