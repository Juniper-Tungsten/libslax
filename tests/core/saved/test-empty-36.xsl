<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:jcs="http://xml.juniper.net/junos/commit-scripts/1.0" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" version="1.0" extension-element-prefixes="jcs slax-ext">
  <xsl:param name="how" select="&quot;and&quot;"/>
  <xsl:template match="/">
    <op-script-results>
      <xsl:value-of select="jcs:output(concat(&quot;This &quot;, $how, &quot; that&quot;))"/>
      <xsl:variable name="slax-temp-arg-1">
        <one/>
      </xsl:variable>
      <xsl:call-template name="test">
        <xsl:with-param xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="one" select="slax-ext:node-set($slax-temp-arg-1)"/>
      </xsl:call-template>
      <xsl:variable name="slax-temp-arg-2">
        <one>two</one>
      </xsl:variable>
      <xsl:call-template name="test">
        <xsl:with-param xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="one" select="slax-ext:node-set($slax-temp-arg-2)"/>
      </xsl:call-template>
      <xsl:variable name="slax-temp-arg-3">
        <one>
          <two>two</two>
        </one>
      </xsl:variable>
      <xsl:call-template name="test">
        <xsl:with-param xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="one" select="slax-ext:node-set($slax-temp-arg-3)"/>
      </xsl:call-template>
    </op-script-results>
  </xsl:template>
  <xsl:template name="test">
    <xsl:param name="one" select="&quot;null&quot;"/>
    <output>
      <xsl:copy-of select="$one"/>
    </output>
  </xsl:template>
</xsl:stylesheet>
