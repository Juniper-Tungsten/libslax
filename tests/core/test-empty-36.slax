version 1.2;

main <op-script-results> {
    var $how = substring("a" _ "n" _ "d", 1);
    call test($one = "This " _ $how _ " that");

    call test($one = <one>);
    call test($one = <one> "two");
    call test($one = <one> { <two> "two"; });
}

template test ($one = "null") {
   <output> { copy-of $one; }
}
