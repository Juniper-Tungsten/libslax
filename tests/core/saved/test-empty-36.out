<?xml version="1.0"?>
<op-script-results>
  <output>
    <one/>
  </output>
  <output>
    <one>two</one>
  </output>
  <output>
    <one>
      <two>two</two>
    </one>
  </output>
</op-script-results>
